CMakeLists.txt

```diff
@@ -4,7 +4,9 @@ cmake_minimum_required(VERSION 3.19...3.25)
 set(CMAKE_CXX_STANDARD  17)
 set(CMAKE_CUDA_STANDARD 17)
 # Latest JAX v0.4.26 no longer supports cuda 11.8
-set(NVHPC_CUDA_VERSION 12.2)
+# By default, build for CUDA 12.2, users can override this with -DNVHPC_CUDA_VERSION=11.8
+set(NVHPC_CUDA_VERSION 12.2 CACHE STRING  "CUDA version to build for" )
```

Comment:
Hum... which nvhpc is this supposed to support? upstream jax is built with cuda 12.3

Reply:
It doesn't matter the version of nvhpc as long as it has the correct CUDA toolkit

Jax is built with 12.3
But on Jeanzay the last module is 12.2
On some HPCs the last is 11.8

I did this to support the maximum number of HPC clusters